torch/jit/annotations.py

```diff
@@ -300,6 +300,8 @@ def try_ann_to_type(ann, loc):
     if inspect.isclass(ann) and is_tensor(ann):
         return TensorType.get()
     if is_tuple(ann):
+        if len(ann.__args__) == 1 and ann.__args__[0] == ():
+            return TupleType([])
```

Comment:
Also add a comment here. It's probably a good idea to comment wherever this empty tuple special casing happens, since it's unintuitive behavior of python. Linking to the definition of the type annotations for tuple, which discusses this could help https://docs.python.org/3/library/typing.html#typing.Tuple